### PR TITLE
Scrapper, Maria Bukina - 22FPL2

### PR DESCRIPTION
**Pull request by MaryBukina**

### CODE REVIEW

**Comment by dmitry-uraev** [comment]
No activity in PR. It is good time to start working on lab assignment

**Comment by dmitry-uraev** [comment]
No activity in PR. It is good time to start working on lab assignment

**Comment by dmitry-uraev** [comment]
nothing to review

**Comment by marina-kaz** [comment]
Nothing to review today.

**Comment by marina-kaz** [comment]
Nothing to review today.

**Comment by marina-kaz** [comment]
Nothing to review today.

**Comment by marina-kaz** [comment]
Nothing to review today.

File: lab_5_scrapper/scrapper.py [hunk 1]
```diff
@@ -18,6 +62,17 @@ def __init__(self, path_to_config: pathlib.Path) -> None:
         Args:
             path_to_config (pathlib.Path): Path to configuration.
         """
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        self.config_dto = self._extract_config_content()
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:67]: 1. Why public?
2. Why store both DTO instance and all its properties?

File: lab_5_scrapper/scrapper.py [hunk 2]
```diff
@@ -100,6 +194,13 @@ def make_request(url: str, config: Config) -> requests.models.Response:
     Returns:
         requests.models.Response: A response from a request
     """
+    period = random.randrange(5)
+    time.sleep(period)
+
+    response = requests.get(url=url, timeout=config.get_timeout(),
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:200]: Make sure you use all the request configurations given you by `scrapper_config.json`

File: lab_5_scrapper/scrapper.py [hunk 3]
```diff
@@ -116,6 +217,9 @@ def __init__(self, config: Config) -> None:
         Args:
             config (Config): Configuration
         """
+        self.urls = []
+        self.config = config
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:221]: Why public?

File: lab_5_scrapper/scrapper.py [hunk 4]
```diff
@@ -127,11 +231,53 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        link = article_bs.find('a', class_="card-big").get('href')
+        url = self.url_pattern + link[len('/articles')::]
+        return url
+
+        #link_tag = article_bs.find('a', class_='card-big')
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:238]: If you don't use it, delete it.

File: lab_5_scrapper/scrapper.py [hunk 5]
```diff
@@ -127,11 +231,53 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
...
+        #if link_tag and 'href' in link_tag.attrs:
+        #    link = link_tag['href']
+        #    url = self.url_pattern + link[len('/articles'):]
+        #    return url
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        #seed_urls = self.get_search_urls()
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:248]: If you don't use it, delete it.

File: lab_5_scrapper/scrapper.py [hunk 6]
```diff
@@ -127,11 +231,53 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
...
+            for article in articles:
+                links = article.find_all('div', class_='articles__list')
+                for link in links:
+                    extracted_url = self._extract_url(link)
+                    if extracted_url and extracted_url not in self.urls:
+                        self.urls.append(extracted_url)
+                    if len(self.urls) >= self.config.get_num_articles():
+                        break
+                if len(self.urls) >= self.config.get_num_articles():
+                    break
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:280]: You don't break from the cycle over `seed_urls`, so even when you've reached desired number of articles, you'll continue to make requests with the next seed URL, am I right?

File: lab_5_scrapper/scrapper.py [hunk 7]
```diff
@@ -160,6 +307,10 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
             article_id (int): Article id
             config (Config): Configuration
         """
+        self.full_url = full_url
+        self.article_id = article_id
+        self.config = config
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:312]: Why public?

**Review by marina-kaz — COMMENTED** [review]
Make sure to delete all commented code before the hand-in.

**Pull request closed without merging**